eras/alonzo/impl/src/Cardano/Ledger/Alonzo/TxWits.hs

```diff
@@ -544,15 +552,30 @@ instance AlonzoEraScript era => DecCBOR (Annotator (RedeemersRaw era)) where
   decCBOR = do
     ifDecoderVersionAtLeast
       (natVersion @9)
-      ( mapTraverseableDecoderA
-          (decodeNonEmptyList decodeAnnElement)
-          (RedeemersRaw . Map.fromList . NE.toList)
+      ( peekTokenType >>= \case
+          TypeMapLenIndef -> decodeMapRedeemers
+          TypeMapLen -> decodeMapRedeemers
+          _ -> decodeListRedeemers
       )
       ( mapTraverseableDecoderA
           (decodeList decodeAnnElement)
           (RedeemersRaw . Map.fromList)
       )
     where
+      decodeRedeemersWith nonEmptyDecoder =
+        mapTraverseableDecoderA
+          nonEmptyDecoder
+          (RedeemersRaw . Map.fromList . NE.toList)
+      decodeMapRedeemers = decodeRedeemersWith $ do
+        (_, xs) <- decodeListLikeWithCount decodeMapLenOrIndef (:) $ \_ -> do
+          ptr <- decCBOR
+          (annData, exUnits) <- decCBOR
+          pure $ (\d -> (ptr, (d, exUnits))) <$> annData
+        case NE.nonEmpty xs of
+          Nothing -> fail "Expected redeemers map to be non-empty"
+          Just neList -> pure $ NE.reverse neList
```

Comment:
Is the `reverse` because passing `:` to  `decodeListLikeWithCount` is reversing the list ?

Reply:
Correct.